Sources/Core/AWSClientRuntime/HttpContextBuilder+Extension.swift

```diff
@@ -83,51 +72,68 @@ extension HttpContext {
-        attributes.set(key: HttpContext.messageSigner, value: messageSigner)
+        attributes.set(key: AttributeKeys.messageEncoder, value: messageEncoder)
+        attributes.set(key: AttributeKeys.messageSigner, value: messageSigner)
 
         // enable the flag
-        attributes.set(key: HttpContext.bidirectionalStreaming, value: true)
+        attributes.set(key: AttributeKeys.bidirectionalStreaming, value: true)
     }
 }
 
 extension HttpContextBuilder {
-
     @discardableResult
-    public func withRegion(value: String?) -> HttpContextBuilder {
-        self.attributes.set(key: HttpContext.region, value: value)
+    public func withCredentialsProvider(value: any CredentialsProviding) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.credentialsProvider, value: value)
         return self
     }
 
     @discardableResult
-    public func withCredentialsProvider(value: any CredentialsProviding) -> HttpContextBuilder {
-        self.attributes.set(key: HttpContext.credentialsProvider, value: value)
+    public func withRegion(value: String?) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.region, value: value)
         return self
     }
 
+    /// Sets the request signature for the event stream operation
+    /// - Parameter value: `String` request signature
     @discardableResult
-    public func withSigningName(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: HttpContext.signingName, value: value)
+    public func withRequestSignature(value: String) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.requestSignature, value: value)
         return self
     }
 
     @discardableResult
-    public func withSigningRegion(value: String?) -> HttpContextBuilder {
-        self.attributes.set(key: HttpContext.signingRegion, value: value)
+    public func withSigningAlgorithm(value: AWSSigningAlgorithm) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.signingAlgorithm, value: value)
         return self
     }
 
     @discardableResult
-    public func withSigningAlgorithm(value: AWSSigningAlgorithm?) -> HttpContextBuilder {
-        self.attributes.set(key: HttpContext.signingAlgorithm, value: value?.rawValue)
+    public func withSigningName(value: String) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.signingName, value: value)
         return self
     }
 
-    /// Sets the request signature for the event stream operation
-    /// - Parameter value: `String` request signature
     @discardableResult
-    public func withRequestSignature(value: String) -> HttpContextBuilder {
-        self.attributes.set(key: HttpContext.requestSignature, value: value)
+    public func withSigningRegion(value: String?) -> HttpContextBuilder {
+        self.attributes.set(key: AttributeKeys.signingRegion, value: value)
         return self
     }
 }
+
+extension AttributeKeys {
```

Comment:
If you put these keys in the `AttributeKey` namespace (i.e. their own type), then you can refer to them without the namespace:
```
        self.attributes.set(key:.signingRegion, value: value)
```
instead of 
```
        self.attributes.set(key: AttributeKeys.signingRegion, value: value)
```

Reply:
https://github.com/smithy-lang/smithy-swift/pull/598#discussion_r1348019020

Reasoning in above link, from previous comment on same issue.